Godeps/_workspace/src/github.com/docker/distribution/registry/storage/layerwriter.go

```diff
@@ -298,8 +336,13 @@ func (lw *layerWriter) validateLayer(dgst digest.Digest) (digest.Digest, error)
 	)
 
 	if lw.resumableDigester != nil {
+		offset := lw.size
+		if offset < lw.offset {
+			// we're probably on NFS where file attributes are cached on client
+			offset = lw.offset
```

Comment:
`resumeHashAt` loads stored hash state with the highest offset lower than or equal to the given one. If the stored one is smaller, the gap will be recomputed from file's content. And since it's our instance which wrote the data up to the `offset`, it should be safe to read it with the same file handle.